File: sentimental-credit/credit.py
from math import pow


def main():
    number = 0
    while number <= 0:
        number = int(input("Number: "))
    length = len(str(number))
    if length == 13 or length == 15 or length == 16:
        find_cc_name(number, length)
    else:
        print("INVALID")


def find_cc_name(number, length):
    copy = number
    sum1, sum2 = 0, 0
    while copy > 0:
        sum1 += copy % 10
        copy //= 10
        digit = (copy % 10) * 2
        sum2 += (digit % 10) + (digit // 10)
        copy //= 10

    sum1 += sum2
    if sum1 % 10 != 0:
        print("INVALID")
    else:
        check = number // pow(10, length - 2)
        if check == 34 or check == 37:
            print("AMEX")
        elif check == 51 or check == 52 or check == 53 or check == 54 or check == 55:
            print("MASTERCARD")
        elif check // 10 == 4:
            print("VISA")
        else:
            print("INVALID")


main()

File: credit/credit.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int find_length(long number);
string find_cc_name(long number, int length);

int main(void)
{

    long number;
    do
    {
        number = get_long("Number: ");
    }
    while (number < 0);

    int length = find_length(number);
    if (length != 13 && length != 15 && length != 16)
    {
        printf("INVALID\n");
    }
    else
    {
        string cc_name = find_cc_name(number, length);
        printf("%s\n", cc_name);
    }
}

int find_length(long number)
{
    int length = 0;
    while (number != 0)
    {
        number /= 10;
        length++;
    }
    return length;
}

string find_cc_name(long number, int length)
{
    int sum1 = 0;
    int sum2 = 0;
    long copy = number;
    do
    {
        sum1 += copy % 10;
        copy /= 10;
        int digit = (copy % 10) * 2;
        sum2 += digit % 10 + digit / 10;
        copy /= 10;
    }
    while (copy > 0);
    sum1 += sum2;
    printf("sum1: %d\n", sum1);
    if (sum1 % 10 != 0)
    {
        return "INVALID";
    }
    else
    {
        int check = number / pow(10, length - 2);
        if (check == 34 || check == 37)
        {
            return "AMEX";
        }
        else if (check == 51 || check == 52 || check == 53 || check == 54 || check == 55)
        {
            return "MASTERCARD";
        }
        else if (check / 10 == 4)
        {
            return "VISA";
        }
        else
        {
            return "INVALID";
        }
    }
}